src/typing.py

```diff
@@ -1949,7 +1949,25 @@ class ValuesView(MappingView[VT_co], extra=collections_abc.ValuesView):
 if hasattr(contextlib, 'AbstractContextManager'):
     class ContextManager(Generic[T_co], extra=contextlib.AbstractContextManager):
         __slots__ = ()
-    __all__.append('ContextManager')
+else:
+    class ContextManager(Generic[T_co]):
+        __slots__ = ()
+
+        def __enter__(self):
+            return self
+
+        @abc.abstractmethod
+        def __exit__(self, exc_type, exc_value, traceback):
+            """Raise any exception triggered within the runtime context."""
+            return None
+
+        @classmethod
+        def __subclasshook__(cls, C):
+            if cls is ContextManager:
+                if (any("__enter__" in B.__dict__ for B in C.__mro__) and
+                        any("__exit__" in B.__dict__ for B in C.__mro__)):
```

Comment:
> I don't think that makes sense here, because this class will only be used in Python < 3.6, and the ``= None`` pattern became generally supported only in 3.6. The collections ABCs don't support anti-registration either before 3.6.

OK, if @gvanrossum agrees with this, then I think this can be merged as is, I don't have any other comments.